lib/std/sort.zig

```diff
@@ -89,23 +93,30 @@ pub fn heapContext(a: usize, b: usize, context: anytype) void {
     }
 }
 
-fn siftDown(a: usize, root: usize, n: usize, context: anytype) void {
-    var node = root;
-    while (true) {
-        var child = a + 2 * (node - a) + 1;
-        if (child >= n) break;
+fn siftDown(a: usize, cur: usize, b: usize, context: anytype) void {
+    var child = (math.mul(usize, cur - a, 2) catch return) + a + 1;
```

Comment:
What exactly happens when this multiplication does overflow? Is there a case where we silently fail to sort fully? It's probably worth a comment in any case.